main/ts_client.h

```diff
@@ -8,11 +8,108 @@
 #define TS_CLIENT_H_
 
 #include <string.h>
+#include <cJSON.h>
+#include <stdint.h>
+
+
+/*
+ * Protocol function codes (same as CoAP)
+ */
+#define TS_GET      0x01
+#define TS_POST     0x02
+#define TS_DELETE   0x04
+#define TS_FETCH    0x05
+#define TS_PATCH    0x07        // it's actually iPATCH
+
+#define TS_PUBMSG   0x1F
+
+/*
+ * Status codes (same as CoAP)
+ */
+
+// success
+#define TS_STATUS_CREATED               0x81
+#define TS_STATUS_DELETED               0x82
+#define TS_STATUS_VALID                 0x83
+#define TS_STATUS_CHANGED               0x84
+#define TS_STATUS_CONTENT               0x85
+
+// client errors
+#define TS_STATUS_BAD_REQUEST           0xA0
+#define TS_STATUS_UNAUTHORIZED          0xA1        // need to authenticate
+#define TS_STATUS_FORBIDDEN             0xA3        // trying to write read-only value
+#define TS_STATUS_NOT_FOUND             0xA4
+#define TS_STATUS_METHOD_NOT_ALLOWED    0xA5
+#define TS_STATUS_REQUEST_INCOMPLETE    0xA8
+#define TS_STATUS_CONFLICT              0xA9
+#define TS_STATUS_REQUEST_TOO_LARGE     0xAD
+#define TS_STATUS_UNSUPPORTED_FORMAT    0xAF
+
+// server errors
+#define TS_STATUS_INTERNAL_SERVER_ERR   0xC0
+#define TS_STATUS_NOT_IMPLEMENTED       0xC1
+
+// ThingSet specific errors
+#define TS_STATUS_RESPONSE_TOO_LARGE    0xE1
+
+/**
+* Struct to hold device information
+* when a new device is connected, the function pointer
+* should be set accordingly to either the serial or CAN send function
+*/
+typedef struct {
+    uint8_t CAN_Address;
+    char *ts_device_id;
+    char *ts_name;
+    char *(*send)(char * req, uint8_t CAN_Address);     //function pointer to send requests to device, abstracting underlying connection
```

Comment:
Maximum line length exceeded.
```suggestion
    // function pointer to send requests to device, abstracting underlying connection
    char *(*send)(char *req, uint8_t CAN_Address);
```